### PR TITLE
Fix #4741,#2514: Datatable cellEditInit AJAX method not firing.

### LINKED ISSUE #4741: cellEditInit: ajax event in dataTable not fired
## 1) Environment
- Tested with 6.2 and the new 7.0 with same result
- Apache Tomee 1.7.4
- Opera

## 2) Expected behavior
cellEditInit-Event should fire. disabled set to "false" not helped 
With rowEditInit it works.

## 3) Actual behavior
Nothing happens.

## 4) Steps to reproduce
Click in one Cell to Edit

## 5) Sample XHTML
Source: https://www.primefaces.org/showcase/ui/data/datatable/edit.xhtml

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:p="http://primefaces.org/ui"
      xmlns:f="http://java.sun.com/jsf/core">
    <h:head>

        <title>DataTable Edit</title>

        <style type="text/css">
            .ui-widget {
                font-size: 80%;
            }  
            .ui-row-editor .ui-row-editor-pencil {
                margin-left:8px;
            }
        </style>
    </h:head>

    <h:body>

        <h:form id="form">
            <p:growl id="msgs" showDetail="true"/>

            

            <p:dialog widgetVar="dialogSave" closable="true">
                DIALOG SAVE
            </p:dialog>

            <p:dialog widgetVar="dialogInit" closable="true">
                DIALOG INIT
            </p:dialog>

            <p:dialog widgetVar="dialogCancel" closable="true">
                DIALOG CANCEL
            </p:dialog>

            <p:dataTable id="cars2" var="car" value="#{dtEditView.cars2}" editable="true" editMode="cell" widgetVar="cellCars">
                <f:facet name="header">
                    Cell Editing
                </f:facet>

                <p:ajax event="cellEdit" listener="#{dtEditView.onCellEdit}" update=":form:msgs" onstart="PF('dialogSave').show()" />
                <p:ajax event="cellEditInit" listener="#{dtEditView.onCellEditInit}" update=":form:msgs" onstart="PF('dialogInit').show()" disabled="false" />
                <p:ajax event="cellEditCancel" listener="#{dtEditView.onCellEditCancel}" update=":form:msgs" onstart="PF('dialogCancel').show()" />

                <p:column headerText="Id">
                    <p:cellEditor>
                        <f:facet name="output"><h:outputText value="#{car.id}" /></f:facet>
                        <f:facet name="input"><p:inputText id="modelInput" value="#{car.id}" style="width:96%"/></f:facet>
                    </p:cellEditor>
                </p:column>

                <p:column headerText="Year">
                    <p:cellEditor>
                        <f:facet name="output"><h:outputText value="#{car.year}" /></f:facet>
                        <f:facet name="input"><p:inputText value="#{car.year}" style="width:96%" label="Year"/></f:facet>
                    </p:cellEditor>
                </p:column>

                <p:column headerText="Brand">
                    <p:cellEditor>
                        <f:facet name="output"><h:outputText value="#{car.brand}" /></f:facet>
                        <f:facet name="input">
                            <h:selectOneMenu value="#{car.brand}" style="width:100%">
                                <f:selectItems value="#{dtEditView.brands}" var="man" itemLabel="#{man}" itemValue="#{man}" />
                            </h:selectOneMenu>
                        </f:facet>
                    </p:cellEditor>
                </p:column>

                <p:column headerText="Color">
                    <p:cellEditor>
                        <f:facet name="output"><h:outputText value="#{car.color}" /></f:facet>
                        <f:facet name="input">
                            <h:selectOneMenu value="#{car.color}" style="width:100%">
                                <f:selectItems value="#{dtEditView.colors}" var="color" itemLabel="#{color}" itemValue="#{color}" />
                            </h:selectOneMenu>
                        </f:facet>
                    </p:cellEditor>
                </p:column>
            </p:dataTable>

        </h:form>

    </h:body>

</html>

## 6) Sample bean
Source: https://www.primefaces.org/showcase/ui/data/datatable/edit.xhtml

package car;

import java.io.Serializable;
import java.util.List;
import javax.annotation.PostConstruct;
import javax.faces.application.FacesMessage;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.ManagedProperty;
import javax.faces.bean.SessionScoped;
import javax.faces.context.FacesContext;
import org.primefaces.event.CellEditEvent;
import org.primefaces.event.RowEditEvent;

@ManagedBean(name = "dtEditView")
@SessionScoped
public class EditView implements Serializable {

    private List<Car> cars1;
    private List<Car> cars2;

    @ManagedProperty("#{carService}")
    private CarService service;

    @PostConstruct
    public void init() {
        cars1 = service.createCars(10);
        cars2 = service.createCars(10);
    }

    public List<Car> getCars1() {
        return cars1;
    }

    public List<Car> getCars2() {
        return cars2;
    }

    public List<String> getBrands() {
        return service.getBrands();
    }

    public List<String> getColors() {
        return service.getColors();
    }

    public void setService(CarService service) {
        this.service = service;
    }

    public void onRowEdit(RowEditEvent event) {
        FacesMessage msg = new FacesMessage("Car Edited", ((Car) event.getObject()).getId());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public void onRowEditInit(RowEditEvent event) {
        FacesMessage msg = new FacesMessage("Edit Init", ((Car) event.getObject()).getId());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public void onRowEditCancel(RowEditEvent event) {
        FacesMessage msg = new FacesMessage("Edit Cancelled", ((Car) event.getObject()).getId());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public void onCellEdit(CellEditEvent event) {
        Object oldValue = event.getOldValue();
        Object newValue = event.getNewValue();

        if (newValue != null && !newValue.equals(oldValue)) {
            FacesMessage msg = new FacesMessage(FacesMessage.SEVERITY_INFO, "Cell Changed", "Old: " + oldValue + ", New:" + newValue);
            FacesContext.getCurrentInstance().addMessage(null, msg);
        }
    }

    public void onCellEditInit(CellEditEvent event) {
        FacesMessage msg = new FacesMessage("Edit Cell Init", event.getOldValue().toString());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public void onCellEditCancel(CellEditEvent event) {
        FacesMessage msg = new FacesMessage("Edit Cell Cancelled", event.getOldValue().toString());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }
}

### PR DESCRIPTION
**Pull request by melloware**
I followed the example of what RowEditInit was doing and made CellEditInit do the same thing.  Tested and everything seems to be working great.  @mertsincan can you please review since this is a DataTable fix.

### CODE REVIEW

**Comment by mertsincan** [comment]
Good PR! Thanks @melloware ;)

File: src/main/resources/META-INF/resources/primefaces/datatable/datatable.js [hunk 1]
```diff
@@ -2495,6 +2501,14 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         }
         else {
             this.showCurrentCell(cell);
+
+            if(this.hasBehavior('cellEditInit')) {
```
Comment (adjiandov) [inline src/main/resources/META-INF/resources/primefaces/datatable/datatable.js:2505]: This solution does not work if the cell is in the last row and you hit tab and the next cell(column) is not editable.

Comment (melloware) [reply src/main/resources/META-INF/resources/primefaces/datatable/datatable.js:2505]: @adjiandov please create an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project and then open a new GitHub ticket with your reproducer.

Comment (adjiandov) [reply src/main/resources/META-INF/resources/primefaces/datatable/datatable.js:2505]: It seems fixed with the latest version. For anyone coming here with older versions, just add the check for 
`if(cellInfo) {
      var ext = {
                    params: [{name: this.id + '_cellInfo', value: cellInfo}]
                };
                this.callBehavior('cellEditInit', ext);
}`

**Pull request merged**